### ISSUE TITLE
Embedded factories don't work without fibrous

### ISSUE THREAD

**Issue by sherrman** [goodeggs/unionized#6]
The examples in the README don't actually work. Here's a crudely hacked together example of it failing to work, and the fibrous example demonstrating that it does.

```javascript
unionized = require('./');
fibrous = require('fibrous');

var pickupFactory = unionized.define(function(done) {
  this.set('pickup.pickupWindow.startAt', '2pm');
  this.set('pickup.pickupWindow.endAt', '4pm');
  this.set('pickup.name', 'San Francisco Ferry Building');
  done();
});
pickupFactory.create(function(err, result) { console.log("pickup factory", result); });

var orderFactory = unionized.define(function(done) {
  this.set('customer', 'Lex Luthor');
  this.set('total', 3.50);
  this.embed('deliveryDetails', pickupFactory, done);
});
orderFactory.create(function(err, result) { console.log("order factory",result); });

fibrous.run(function () {
  var pf2 = unionized.define(fibrous(function() {
    this.set('pickup.pickupWindow.startAt', '2pm');
    this.set('pickup.pickupWindow.endAt', '4pm');
    this.set('pickup.name', 'San Francisco Ferry Building');
  }));

  var orderFactory2 = unionized.define(fibrous(function() {
    this.set('customer', 'Lex Luthor');
    this.set('total', 3.50);
    this.sync.embed('deliveryDetails', pf2);
  }));
  console.log('fib of2', orderFactory2.sync.create());

});
```

**Comment by demands** [comment]
This is fixed in `4.0.0` &mdash; the tests don't use fibrous anymore!

**Comment by sherrman** [comment]
Beautiful! Thanks you two!
